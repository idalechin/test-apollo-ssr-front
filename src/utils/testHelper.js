import React from "react";
import { ApolloClient } from "apollo-client";
import { ApolloProvider } from "react-apollo-hooks";
import { InMemoryCache } from "apollo-cache-inmemory";
import { MockLink } from "apollo-link-mock";

export const MockClient = ({ mocks, children }) => {
	function createClient(mocks) {
		return new ApolloClient({
			cache: new InMemoryCache(),
			link: new MockLink(mocks)
		});
	}

	const client = createClient(mocks);

	return (
		<ApolloProvider client={client}>
			{children}
		</ApolloProvider>
	);
};
