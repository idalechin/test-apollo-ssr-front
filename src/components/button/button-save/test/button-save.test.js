import React from "react";
import { shallow, mount } from "enzyme";
import { render, fireEvent, cleanup } from "react-testing-library";
import SaveButton, { TOGGLE_VENDOR_SAVE, VENDOR_IS_SAVED } from "../index";
import renderer from "react-test-renderer";
import { MockClient } from '../../../../utils/testHelper'

const className = "btn-test";
const size = "small";
const id = "18";
const mocks = [
	{
		request: {
			query: VENDOR_IS_SAVED,
			variables: {id}
		},
		result: {
			data: {
				vendorIsSaved: { id, value: true }
			}
		}
	}
];

describe("<SaveButton />", () => {
	it("Test renders right", () => {
		const btn = renderer
			.create(
				<MockClient mocks={mocks}>
					<SaveButton
						type="vendor"
						id={id}
						className={className}
						size={size}
						border
					/>
				</MockClient>
			)
			.toJSON();
		expect(btn).toMatchSnapshot();
	});
});
