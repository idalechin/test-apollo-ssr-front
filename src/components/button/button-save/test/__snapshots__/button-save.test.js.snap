// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveButton /> Test renders right 1`] = `
<button
  className="btn btn-test btn--secondary-border btn--icon btn--like btn--small-like  "
  onClick={[Function]}
>
  <svg
    className="btn__icon"
    viewBox="undefined"
  >
    <use
      xlinkHref="#undefined"
    />
  </svg>
  <span
    className="btn__text"
  />
</button>
`;
